specification/appcomplianceautomation/resource-manager/Microsoft.AppComplianceAutomation/preview/2022-11-16-preview/appcomplianceautomation.json

```diff
@@ -0,0 +1,1374 @@
+          "type": "string"
+        }
+      }
+    },
+    "SnapshotResourceList": {
+      "description": "Object that includes an array of resources and a possible link for next set.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of the snapshots",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/SnapshotResource"
+          }
+        },
+        "nextLink": {
+          "description": "The URL the client should use to fetch the next page (per server side paging).\r\nIt's null for now, added for future use.",
+          "type": "string"
+        }
+      }
+    },
+    "ResourceItem": {
+      "type": "object",
+      "description": "Resource Id.",
+      "properties": {
+        "subscriptionId": {
+          "description": "The subscription Id of this resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "resourceGroup": {
+          "description": "The resource group name of this resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "resourceType": {
+          "description": "The resource type of this resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "resourceId": {
+          "description": "The resource Id.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "DownloadResponse": {
+      "description": "Object that includes all the possible response for the download operation.",
+      "type": "object",
+      "properties": {
+        "resourceList": {
+          "description": "List of the reports",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ResourceItem"
+          },
+          "x-ms-identifiers": []
+        },
+        "complianceReport": {
+          "description": "List of the compliance result",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ComplianceReportItem"
+          },
+          "x-ms-identifiers": []
+        },
+        "compliancePdfReport": {
+          "description": "compliance pdf report",
+          "type": "object",
+          "readOnly": true,
+          "properties": {
+            "sasUri": {
+              "description": "uri of compliance pdf report",
+              "type": "string",
+              "readOnly": true
+            }
+          }
+        },
+        "complianceDetailedPdfReport": {
+          "description": "compliance detailed pdf report",
+          "type": "object",
+          "readOnly": true,
+          "properties": {
+            "sasUri": {
+              "description": "uri of compliance detailed pdf report",
+              "type": "string",
+              "readOnly": true
```

Comment:
could this contain secrets in the query string like sig values etc ? if so please mark as x-ms-secret

Reply:
@raosuhas Yes, I think we can add the "**x-ms-secret**".

Reply:
Hi @raosuhas ~
Actually the sasUri is not a secret, we don't get the value from users' input. Our service generate the sasUri, and return the value in the response.
When user get the sasUri, they can use the sasUri to download a file within 10 minutes. After 10 minutes, the uri will expire.
So add a "x-ms-secret", but set the value to false temporarily.
See [here](https://github.com/Azure/azure-rest-api-specs/pull/21152/files#diff-16cc0066db31d47ee8b4f0e9385ece034881677d74a14b4d48790444b24bda4dL1183).

Reply:
A sasUri is a secret, even when it is not from the user's input, because it is a secret value that allows a privileged operation, in this case, downloading a file. You should be able to mark it as x-ms-secret as long as it is only returned from POST requests, and not GET requests.